docs: small updates for latest pattern compliance (#6227)
snapcore/snapcraft: docs/index.rst

.. meta::
    :description: Documentation for Snapcraft, the tool for packaging software into the snap container format.

Snapcraft
=========

**Snapcraft** is the tool for packaging software into the snap container format.

> It builds and bundles Linux software of all kinds and sources, and is operated through a
> command-line interface.

Reviewer: ```suggestion
It builds and bundles software of all kinds and sources, and is operated from the
command line.
```

> Linux software

I assume you mean that it's packaging software _for_ Linux systems, but this sounds like it's able to package software _for Linux_. If that didn't make any sense, ignore me. It's late.

Reviewer 2: I think it's reasonable to specify "Linux software", given that snapd is only available on Linux
